backend/inspirehep/tools/utils.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+
+import re
+
+from werkzeug.utils import secure_filename
+
+from inspirehep.search.api import LiteratureSearch
+
+FORMAT_TO_SOURCE_FIELD = {
+    "latex_eu": "_latex_eu_display",
+    "latex_us": "_latex_us_display",
+    "bibtex": "_bibtex_display",
+}
+
+
+def get_references(f):
+    """Extract references from LaTeX string (whole file)"""
+
+    references = []
+    cstrip = re.compile(r"(?<!\\)%.*$", re.M)
+
+    for num, line in enumerate(f, 1):
+        if cstrip.match(line):
+            continue
+        cites = re.findall(r"\\cite\s*\{(.*?)\}", line, re.DOTALL)
+        for ref_line in cites:
+            ref_list = ref_line.split(",")
+            for one_ref in ref_list:
+                one_ref = re.sub(r"\s", "", one_ref)
+                if re.match(r"^#\d{1,2}$", one_ref):
+                    continue
+                if one_ref not in references:
+                    references.append({"value": one_ref, "line": num})
```

Comment:
I would:
* use a tuple here, so you can unpack the tuple easily while iterating
* transform this into a generator (documenting it in the function name/docstring), so you don't need to build the whole reference list ahead of time. Remove L25 and
```suggestion
                    yield (one_ref, num)
```

Reply:
The problem with this is that I need to check if the reference is repeated.